src/restapi/fConnect.ts

```diff
@@ -1,13 +1,15 @@
 import {Request, Response} from 'express';
 
+import {AppContextWithBot} from 'types/apps';
+
 import {newOKCallResponseWithMarkdown} from '../utils/call_responses';
 
 import {newConfigStore} from '../store';
 
-import {Routes, createOAuthState, contextFromRequest} from '../utils';
+import {Routes, createOAuthState} from '../utils';
 
 export async function fConnect(req: Request, res: Response): Promise<void> {
-    const context = contextFromRequest(req);
+    const context: AppContextWithBot = req.body.context;
```

Comment:
1/5 that I like the helper here.

Reply:
I don't mind adding this helper func back, but it will need to change.  The AppContext will change depending on the expand values.  I think I'd need to add another input that defined the `type` of context that is expected and what should be returned from the function.

Or.. I could create several helpers that accept a `req`, but return and App Context that is appropriate.  For example 
`contextWithBotFromRequest(req)`
`contextWithPostFromRequest(req)`

Reply:
By not having the helpers, we can just cast the type when defining the variable.

`const context: AppContextWithBot: = req.body.context`

Reply:
I went with creating concrete types that are now used by the call functions.  I could still create the helper functions which would be good for checking the existence of the values.